xbmc/windowing/android/WinSystemAndroidGLESContext.cpp

```diff
@@ -126,9 +126,6 @@ void CWinSystemAndroidGLESContext::PresentRenderImpl(bool rendered)
     for (std::vector<IDispResource *>::iterator i = m_resources.begin(); i != m_resources.end(); ++i)
       (*i)->OnResetDisplay();
   }
-  if (!rendered)
-    return;
```

Comment:
you should not call SwapBuffers if nothing was rendered. the content of the backbuffer is undefined after swap

Reply:
There is no other way to get an exact swap timestamp, can you give me an alternative way?

Reply:
I tred to get VSYNC out of other sorces (Choreographer for example) but they are all useless.
IMO (e)GL should be intelligent enough to see that nothing has rendered

Reply:
I use this for X11 https://chromium.googlesource.com/chromium/src/gpu/+/master/GLES2/extensions/CHROMIUM/EGL_CHROMIUM_sync_control.txt

Reply:
This CHROMIUM extension is not available in android GL(ES) implementation (not in toolchain, not on the devices I tested)
There is NO other halfways stable vsync signal available in android, the Choreographer one is useless.

Until we have tested several different devices, and all look fine, I think its a good idea to let it go as it is.

Painting transparent is not working as it breaks smartredraw - the goal is to not paint anything, but have the VSYNC.
Seenms to work fine on all android devices I tested so far.

Reply:
Does not change the fact that it is wrong. Unless EGL_BUFFER_PRESERVED is set and implemented back buffer is undefined after a swap. At least you have to call something like glClear on the buffer before swapping.

Reply:
I trust you, but it works well and exactly as expected / required for the needs we have for android.
If this will lead later / on any devices to issues, it will be most likely me fixing it.

Reply:
glClear does not work for smartredraw, because you overwrite GUI with back-buffer

Reply:
What issue do you get if you add the early return back?

Reply:
100% CPU and ugly 25/50. Somehow I have th throttle the Renderloop